packages/account/src/providers/transaction-response/transaction-response.ts

```diff
@@ -93,22 +130,23 @@ export class TransactionResponse {
   /** Gas used on the transaction */
   gasUsed: BN = bn(0);
   /** The graphql Transaction with receipts object. */
-  gqlTransaction?: GqlTransaction;
-
+  private gqlTransaction?: GqlTransaction;
+  private request?: TransactionRequest;
+  private status?: GqlStatusChangeSubscription['statusChange'];
   abis?: JsonAbisFromAllCalls;
-  /** The expected status from the getTransactionWithReceipts response */
-  private expectedStatus?: GqlTransactionStatusesNames;
 
   /**
    * Constructor for `TransactionResponse`.
    *
-   * @param id - The transaction ID.
+   * @param tx - The transaction ID or TransactionRequest.
    * @param provider - The provider.
    */
-  constructor(id: string, provider: Provider, abis?: JsonAbisFromAllCalls) {
-    this.id = id;
+  constructor(tx: string | TransactionRequest, provider: Provider, abis?: JsonAbisFromAllCalls) {
```

Comment:
Not a fan of the type overload here, dealing with just the ID was fine IMO.

Reply:
Nevermind @petertonysmith94 taught me the error of my ways, we reusing the request as we have the fields, otherwise we fetch, of course 🤦🏻

Reply:
The alternative would be:
```ts
constructor(
  id: string, 
  provider: Provider, 
  abis?: JsonAbiFromAllCalls, 
  private request?: TransactionRequest) { ... }

// Provider.sendTransaction
return new TransactionResponse(
  transactionRequest.getTransactionId(this.getChainId(),
  this,
  abis,
  transactionRequest)
```

The type union makes the two use cases of `TransactionResponse` explicit; you either specify the id and we'll fetch everything else, or you give us your `TransactionRequest` of the transaction you just sent and we'll fetch only the necessary data.

Reply:
Makes sense, LGTM 🫡